syntax = "proto3";

package istio.envoy.config.filter.http.alpn.v2alpha1;

import "udpa/annotations/status.proto";

option java_package = "io.envoyproxy.istio.envoy.config.filter.http.alpn.v2alpha1";
option java_outer_classname = "AlpnProto";
option java_multiple_files = true;
option go_package = "github.com/envoyproxy/go-control-plane/contrib/istio/envoy/config/filter/http/alpn/v2alpha1";
option (udpa.annotations.file_status).package_version_status = ACTIVE;

// [#protodoc-title: ALPN HTTP filter]
//
// ALPN HTTP filter from Istio.
//
// [#extension: envoy.filters.http.alpn]

// FilterConfig is the config for Istio-specific filter.
message FilterConfig {
  // Upstream protocols
  enum Protocol {
    HTTP10 = 0;
    HTTP11 = 1;
    HTTP2 = 2;
  }

  message AlpnOverride {
    // Upstream protocol
    Protocol upstream_protocol = 1;

    // A list of ALPN that will override the ALPN for upstream TLS connections.
    repeated string alpn_override = 2;
  }

  // Map from upstream protocol to list of ALPN
  repeated AlpnOverride alpn_override = 1;
}
